static/app/views/preprod/components/visualizations/appSizeLegend.tsx

```diff
@@ -0,0 +1,100 @@
+import {useTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import {Flex} from 'sentry/components/core/layout';
+import {getAppSizeCategoryInfo} from 'sentry/views/preprod/components/visualizations/appSizeTheme';
+import {TreemapType, type TreemapElement} from 'sentry/views/preprod/types/appSizeTypes';
+
+interface AppSizeLegendProps {
+  onToggleCategory: (category: TreemapType) => void;
+  root: TreemapElement;
+  selectedCategories: Set<TreemapType>;
+}
+
+export function AppSizeLegend({
+  root,
+  selectedCategories,
+  onToggleCategory,
+}: AppSizeLegendProps) {
+  const theme = useTheme();
+  const appSizeCategoryInfo = getAppSizeCategoryInfo(theme);
+
+  const categoryTypes = new Set<TreemapType>();
+
+  function collectCategories(element: TreemapElement) {
+    if (element.type) {
+      categoryTypes.add(element.type);
+    }
+    if (element.children) {
+      element.children.forEach(child => collectCategories(child));
+    }
+  }
+
+  collectCategories(root);
+
+  const sortedCategories = Array.from(categoryTypes).sort((a, b) => {
+    const categoryA = appSizeCategoryInfo[a];
+    const categoryB = appSizeCategoryInfo[b];
+    return (categoryA?.displayName || a).localeCompare(categoryB?.displayName || b);
+  });
+
+  return (
+    <Flex gap="xs" wrap="wrap" align="center" justify="end">
+      {sortedCategories.map(categoryType => {
+        const categoryInfo =
+          appSizeCategoryInfo[categoryType] ?? appSizeCategoryInfo[TreemapType.OTHER];
+        if (!categoryInfo) {
+          return null;
+        }
+        const isActive =
+          selectedCategories.size === 0 || selectedCategories.has(categoryType);
```

Comment:
Should we just seed the selected categories with all categories rather than having the size === 0 logic?